src/tpm.rs

```diff
@@ -23,3 +42,53 @@ pub(crate) fn get_tpm2_ctx() -> Result<tss_esapi::Context, tss_esapi::Error> {
     let tcti = Tcti::from_str(tcti_path)?;
     unsafe { Context::new(tcti) }
 }
+
+/*
+ * Input: None
+ * Return:
+ *   TPM Vendor Type
+ *   tss_esapi::Error
+ *
+ * get_tpm_vendor will retrieve the type of TPM. This allows us
+ * to perform opertions, such as understand the host is using a
+ * software TPM as opposed to hardware TPM.
+ */
+pub fn get_tpm_vendor() -> Result<String, tss_esapi::Error> {
+    let mut ctx = get_tpm2_ctx()?;
+
+    let mut allprops = HashMap::new();
+    let (capabs, more) = ctx.get_capabilities(
+        TPM2_CAP_TPM_PROPERTIES,
+        TPM2_PT_MANUFACTURER,
+        80,
+    )?;
+
+    if capabs.capability != TPM2_CAP_TPM_PROPERTIES {
+        panic!("Invalid property returned");
```

Comment:
nit: May be worth noting what was expected and/or what was received.

Reply:
Well, this `TPM2_CAP_TPM_PROPERTIES` expectation is because a few lines up, it's asking for `get_capabilities(TPM2_CAP_TPM_PROPERTIES, ...)`, so this is purely a "Huh, why did the TPM return something incorrect".